docs/design/rubygems.md

```diff
@@ -0,0 +1,528 @@
+GEM
+ remote: https://rubygems.org/
+ specs:
+  rails (6.1.4)
+    # transitive dependencies
+    actioncable (= 6.1.4)
+    actionmailbox (= 6.1.4)
+    actionmailer (= 6.1.4)
+    actionpack (= 6.1.4)
+    actiontext (= 6.1.4)
+    actionview (= 6.1.4)
+    activejob (= 6.1.4)
+    activemodel (= 6.1.4)
+    activerecord (= 6.1.4)
+    activestorage (= 6.1.4)
+    activesupport (= 6.1.4)
+    bundler (>= 1.15.0)
+    railties (= 6.1.4)
+    sprockets-rails (>= 2.0.0)
+```
+
+#### Git dependencies
+Example of a Git dependency in the `Gemfile.lock` file:
+
+```
+GIT
+  remote: https://github.com/porta.git
+  revision: 779beabd653afcd03c4468e0a69dc043f3bbb748
+  branch: main
+  specs:
+    porta (2.14.1)
+```
+
+#### Path dependencies
+Example of a path dependency in the `Gemfile.lock` file:
+
+```
+PATH
+  remote: some/pathgem
+  specs:
+    pathgem (0.1.0)
+```
+
+All path dependencies must be in the project directory. Bundler
+[does not copy](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/source/path.rb#L83) those
+dependencies that are already within the root directory of the project.
+
+#### Plugins
+Installing a plugin, even when on a folder that is a Bundler project, doesn't seem to affect the `Gemfile.lock`. The
+plugin seems to be installed by default in the `$PWD/.bundle/`. The `Gemfile.lock` does have a section for plugins,
+though, so further investigation would be needed. This initial investigation was done with the plugins listed under
+[Known Plugins](https://bundler.io/guides/plugins.html).
+
+*Don't confuse Bundler plugins with [RubyGems plugins](https://guides.rubygems.org/plugins/). The latter are meant to
+extend the functionality of `gem` itself, and don't seem to have any impact on Bundler directly.*
+
+
+### Platforms
+Some gems may contain pre-compiled binaries that provide native extensions to the Ruby package. Any gem declared in the
+`Gemfile` can be limited to specific
+[platforms](https://bundler.io/v2.5/man/gemfile.5.html#PLATFORMS), making Bundler ignore it in case the project is
+being built on a non-matching platform:
+
+```ruby
+gem "nokogiri",   platforms: [:windows_31, :jruby]
+```
+
+Here's an example of how a the `PLATFORM` section looks like in the `Gemfile.lock`:
+
+```
+PLATFORMS
+  arm64-darwin-20
+  arm64-darwin-21
+  arm64-darwin-22
+  ruby
+  x86_64-darwin-18
+  x86_64-darwin-20
+  x86_64-darwin-21
+  x86_64-darwin-22
+  x86_64-linux
+```
+
+In case a user wants to force all the binaries to be compiled from source, the `BUNDLE_FORCE_RUBY_PLATFORM` environment
+variable can be used.
+
+### Dev dependencies
+When adding a Gem into a Gemfile, the user might opt to nest them under a specific
+[group](https://bundler.io/guides/groups.html). The name of the group can be any string, but the usual groups tend to
+be common labels such as `:test`, `:development` or `:production`.
+
+Here's how it looks like in a `Gemfile`:
+
+```ruby
+# :default group
+gem 'nokogiri'
+
+group :test do
```

Comment:
This (and the 'gemspec' file above), seems to be executable Ruby code. If that is the case, are there any restrictions on what is allowed? I don't think that cachi2 wants to allow just anything at all (c.f. Yarnberry :skull:)

Reply:
Ignore, I see that this is addressed later in the doc